### DISCUSSION TITLE
errors from jules.5.x (5.6)

### DISCUSSION THREAD

**Discussion by jachungithub** [NASA-LIS/LISF#1072] [Q&A] [answered]
Hi,

I am testing the jules.5.x (specifically 5.6) in the LISF-public-7.3.3. However, there are no ldt.config in the testcases/surfacemodels/land/jules.5.x or other jules versions. I just guessed some of ldt configures from the lis.config in the testcases (The lis.config says only two sources are coming from ldt: i.e., Landmask data source: "LDT", and Landcover data source: "LDT", the others are set to "none" in lis.config in the testcases.). However, my test-run of the jules.5.x (specifically 5.6) were failed. I attached:
> 1.  ldtlog.0000
> 2. out.o116375 (an error/ouput log file)
> 3. ldt.config
> 4. configure.lis (For just in case...it's almost same as configure.ldt. It's hard to get the configure.ldt now for some security issue.)

Currently, with similar configurations (i.e., ldt.config), LDT for Noah.3.6 has been successfully run. 

Additionally, 
I am wondering if I can get **_ldt.config_** and **_lis.config_** for any **_JULES_** versions and other configuration files (e.g., **_forcing_variables.txt_**)
It would be greatly appreciated.

Thanks a lot,
Jong

[ldt.config_errors.zip](https://github.com/NASA-LIS/LISF/files/8572306/ldt.config_errors.zip)

**Comment by dmocko** [comment, marked as answer]
Hi @jachungithub 

Have you followed the JULES coupling instructions - including downloading the JULES code from the UK MetOffice - that are detailed here?

https://github.com/NASA-LIS/LISF/blob/master/lis/surfacemodels/land/jules.5.x/COUPLING_INSTRUCTIONS.adoc

You are seeing that specific error because you listed "JULES" as the `Land surface model:` in the ldt.config file.

If you check the LDT Users' Guide, "JULES" is not a valid entry for this config option, which is why you get that error.

"JULES.5.0" is the only version of JULES supported by LDT, as listed in the Users' Guide.

If you don't have permission to get the code from the UK MetOffice, then you will not be able to use JULES in LDT/LIS.

If you do have permission, can you please let us know of any connections to the UK MetOffice for working with JULES in LISF?

David

**Reply by jachungithub** [reply]
Thanks a lot,
First of all, I have the permission, received the jules source files, and compiled LIS as Guide. However, I don't have any direct connections to the UK MetOffice. I just requested the source codes via the web-site. Anyhow, I changed "JULES" to "JULES.5.0", but didn't work. I found there are several additional files from "JULES50_parmsMod.F90" (~ldt/params/JULES/JULES50_parmsMod.F90). For examples, JULES ancillary file, JULES WHS soil parmater file etc. Where can I get those files? or some of configurations in ldt.config? 
One more question, if I successfully generate the "lis_input.d01.nc" from JULES.5.0, can I use this file for JULES.5.x (specifically, JULES.5.6.)?
Thanks a lot, again.

**Reply by dmocko** [reply]
Hi @jachungithub 

A sample ldt.config file for JULES.5.0 can be found here:

https://github.com/NASA-LIS/LISF/blob/master/lis/configs/557WW-7.4-FOC/NRT_GLOBAL/for_discover/ldt.config.global.jules50

You will need to generate the JULES ancillary file yourself, using various netCDF tools.  You should be able to get the source ancillary dataset from the UK Met Office JULES site.  The "ncdump -h" of the JULES ancillary file after you create it should look similar to:

```
netcdf cap2ldt_ps41 {
dimensions:
        lon = 2560 ;
        lat = 1920 ;
        pft = 9 ;
variables:
        float landmask(lat, lon) ;
        float landcover(pft, lat, lon) ;
        float b(lat, lon) ;
        float albedo(lat, lon) ;
        float carbon(lat, lon) ;
        float smvcst(lat, lon) ;
        float smvccl(lat, lon) ;
        float smvcwt(lat, lon) ;
        float satcon(lat, lon) ;
        float sathh(lat, lon) ;
        float hcap(lat, lon) ;
        float hcon(lat, lon) ;
        float mean_ti(lat, lon) ;
        float std_ti(lat, lon) ;
        float fexp(lat, lon) ;
}
```

This is for a grid of 2560 by 1920, corresponding to the global domain as shown in the sample ldt.config file linked above.

I believe that you can use the "lis_input" file from JULES.5.0 for any version of JULES.

**Reply by jachungithub** [reply]
Thanks a lot.
